src/app/snapp_test_transaction/snapp_test_transaction.ml

```diff
@@ -0,0 +1,1247 @@
+    let open Deferred.Let_syntax in
+    let%bind receivers = receivers in
+    let amount =
+      List.fold ~init:Currency.Amount.zero receivers ~f:(fun acc (_, a) ->
+          Option.value_exn (Currency.Amount.add acc a))
+    in
+    let%bind keypair = Util.keypair_of_file keyfile in
+    let spec =
+      { Transaction_snark.For_tests.Spec.sender = (keypair, nonce)
+      ; fee
+      ; receivers
+      ; amount
+      ; snapp_account_keypair = None
+      ; memo = Util.memo memo
+      ; new_snapp_account = false
+      ; snapp_update = Party.Update.dummy
+      ; current_auth = Permissions.Auth_required.Proof
+      ; call_data = Snark_params.Tick.Field.zero
+      ; events = []
+      ; sequence_events = []
+      }
+    in
+    let parties = Transaction_snark.For_tests.multiple_transfers spec in
+    Util.print_snapp_transaction parties ;
+    let%map () =
+      if debug then gen_proof parties ~snapp_account:None else return ()
+    in
+    ()
+  in
+  let read_key_and_amount count =
+    let read () =
+      let open Deferred.Let_syntax in
+      Core.printf "Receiver Key:%!" ;
+      match%bind Reader.read_line (Lazy.force Reader.stdin) with
+      | `Ok key -> (
+          let pk =
+            Signature_lib.Public_key.Compressed.of_base58_check_exn key
+          in
+          Core.printf !"Amount:%!" ;
+          match%map Reader.read_line (Lazy.force Reader.stdin) with
+          | `Ok amt ->
+              let amount = Currency.Amount.of_formatted_string amt in
+              (pk, amount)
+          | `Eof ->
+              failwith "Invalid amount" )
+      | `Eof ->
+          failwith "Invalid key"
+    in
+    let rec go ?(prompt = true) count keys =
+      if count <= 0 then return keys
+      else if prompt then (
+        Core.printf "Continue?[y/n]\n%!" ;
+        match%bind Reader.read_line (Lazy.force Reader.stdin) with
+        | `Ok r ->
+            if String.(equal (lowercase r) "y") then
+              let%bind key = read () in
+              go (count - 1) (key :: keys)
+            else return keys
+        | `Eof ->
+            failwith "Invalid input" )
+      else
+        let%bind key = read () in
+        go (count - 1) (key :: keys)
+    in
+
+    Core.printf "Enter at most %d receivers (base58 encoding) and amounts\n%!"
+      count ;
+    let%bind ks = go ~prompt:false 1 [] in
+    go (count - 1) ks
+  in
+  Command.(
+    let open Let_syntax in
+    Command.async
+      ~summary:
+        "Generate a snapp transaction that makes multiple transfers from one \
+         account"
+      (let%map keyfile, fee, nonce, memo, debug = Flags.common_flags in
+       let fee = Option.value ~default:Flags.default_fee fee in
+       if Currency.Fee.(fee < Flags.min_fee) then
+         failwith
+           (sprintf "Fee must at least be %s"
+              (Currency.Fee.to_formatted_string Flags.min_fee)) ;
+       let receivers = read_key_and_amount 10 in
```

Comment:
Hard-coded `10` here feels less than ideal.

Reply:
Giving this a name, `num_attempts`.